src/keymap.rs

```diff
@@ -13,6 +14,58 @@ use crate::tty::{RawReader, Term, Terminal};
 /// The number of times one command should be repeated.
 pub type RepeatCount = usize;
 
+/// Custom dynamic action
+#[derive(Clone)]
+pub struct Action {
+    pub name: String,
+    /// Takes the currently edited `line` with the cursor `pos`ition and
+    /// returns the command to be performed or `None` to perform the default
+    /// one.
+    pub action: fn(line: &str, pos: usize, ctx: &ActionContext) -> Option<Cmd>,
+}
+
+impl std::fmt::Debug for Action {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Action").field("name", &self.name).finish()
+    }
+}
+
+impl PartialEq for Action {
+    fn eq(&self, other: &Self) -> bool {
+        self.name == other.name
+    }
+}
+
+pub struct ActionContext<'r> {
```

Comment:
You've made the struct public but it's isn't exposed anywhere so no documentation gets generated for it.

Reply:
My bad.